### PR TITLE
UI: Warn when no gc rules are defined for long living repos

### LINKED ISSUE #5399: UI: Warn when a repository has no GC rules configured
lakeFS should provide a (dismissible) warning if a repository has no GC rules configured.
Ideally, to make this warning less intrusive, should probably only warn if a repository has been alive for a long enough period of time (or find some other heuristic based on usage/commits/??) 

I suggest doing so in the lakeFS UI as a first step.

I'll open a separate issue for warning on GC rules configured but no recent GC runs completed (I think this one requires a bit more work as the lakeFS API does not expose an API for GC runs at the moment).

### PR DESCRIPTION
**Pull request by ozkatz**
Closes #5399

Example of a warning:

image

Dismissing the warning is persistent in the browser (for the current repository)

### CODE REVIEW

File: webui/src/pages/repositories/repository/objects.jsx [hunk 1]
```diff
@@ -364,6 +366,49 @@ const ReadmeContainer = ({repo, reference, path='', refreshDep=''}) => {
     );
 }
 
+const NoGCRulesWarning = ({ repoId }) => {
+    const storageKey = `show_gc_warning_${repoId}`;
+    const [show,setShow] = useState(window.localStorage.getItem(storageKey) !== "false")
```
Comment (Jonathan-Rosenberg) [inline webui/src/pages/repositories/repository/objects.jsx:371]: nit:
```suggestion
    const [show, setShow] = useState(window.localStorage.getItem(storageKey) !== "false")
```

File: webui/src/pages/repositories/repository/settings/layout.jsx [hunk 2]
```diff
@@ -32,12 +32,12 @@ export const SettingsLayout = ({ children, activeTab }) => {
                                 </Nav>
                                 <Nav variant="pills" className="flex-column">
                                     <Link component={Nav.Link} href={{pathname: '/repositories/:repoId/settings/retention', params: {repoId }}} active={activeTab === 'retention'}>
-                                        Retention
+                                        Garbage Collection
```
Comment (Jonathan-Rosenberg) [inline webui/src/pages/repositories/repository/settings/layout.jsx:35]: 🥳

**Review by Jonathan-Rosenberg — APPROVED** [review]
Thanks!

**Review by talSofer — APPROVED** [review]
Thanks @ozkatz!!

**Pull request merged**